posthog/temporal/usage_reports/run_usage_reports.py

```diff
@@ -0,0 +1,197 @@
+import json
+from temporalio import activity, workflow
+from datetime import datetime, timedelta
+from typing import Optional
+import dataclasses
+import structlog
+import logging
+
+from dateutil import parser
+from django.conf import settings
+from posthog.temporal.common.base import PostHogWorkflow
+
+from posthog.utils import (
+    get_instance_region,
+    get_previous_day,
+)
+
+from posthog.tasks.usage_report import (
+    get_instance_metadata,
+    get_ph_client,
+    _get_all_org_reports,
+    _get_full_org_usage_report,
+    _get_full_org_usage_report_as_dict,
+    has_non_zero_usage,
+    _queue_report,
+    capture_report,
+    OrgReport,
+)
+
+logger = structlog.get_logger()
+logging.basicConfig(level=logging.INFO)
+
+
+@dataclasses.dataclass
+class RunUsageReportsInputs:
+    at: Optional[str] = None
+    skip_capture_event: Optional[bool] = False
+
+
+@dataclasses.dataclass
+class QueryUsageReportsInputs:
+    at: Optional[str] = None
+
+
+@dataclasses.dataclass
+class QueryUsageReportsResult:
+    org_reports: dict[str, OrgReport]
+    period: tuple[datetime, datetime]
+
+
+@dataclasses.dataclass
+class SendUsageReportsInputs:
+    org_reports: dict[str, OrgReport]
+    period: tuple[datetime, datetime]
+    skip_capture_event: Optional[bool] = False
+    at: Optional[str] = None
+
+
+@activity.defn(name="query-usage-reports")
+async def query_usage_reports(
```

Comment:
it may be worth using our logger from `posthog/temporal/common/logger.py` (`bind_temporal_worker_logger()`). It logs out to `log_entries` in clickhouse. You'll just need to configure `get_temporal_context()` for your workflow

Reply:
We don't have a team id, so we can chat about this in our call.